src/intel/RoomIntel.ts

```diff
@@ -456,7 +458,46 @@ export class RoomIntel {
 			}
 		}
 	}
+	private static autoPoison(room: Room) {
+		
+		if(!Memory.settings.autoPoison.enabled || !room || getAllColonies().length == Game.gcl.level) {
+			return;
+		}
 
+		// poison X rooms at a time
+		let count = 0;
+		for (const colony of getAllColonies()) {
+			if(Game.flags['POISON-'+colony.name]) {
+				count++;
+			}
+			if (count >= Memory.settings.autoPoison.concurrent) {
+				return;
+			}
+		}
+		
+		if (!(room.controller && room.controller.level == 0 && room.controller.reservation == undefined 
+			&& !room.storage && !room.terminal // do not include rooms with storage/terminal, i might want to loot
+			&& room.hostiles.length == 0)
+			|| DirectiveOutpost.isPresent(room.controller.pos,'pos')) {  // do not poison own outPost!
+			return; // not a valid room to poison;
+		} 
+		
+		const walkableControllerPositions = _.filter(room.controller.pos.neighbors, pos => pos.isWalkable(true));
+		if (walkableControllerPositions.length == 0) {
+			return; // already poisoned.
+		}
+		
+		const colonies = getAllColonies().filter(colony => colony.level > 6);
+		for (const colony of colonies) {
+			const route = Game.map.findRoute(colony.room, room);
```

Comment:
First do a check against linear distance. I added it to the other ones. It stops us trying to find a path from rooms that are 20+ away, saves some cpu.

Secondly, you do a filter against colony level here, and also in the directive, and they are different levels.